packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -90,6 +93,34 @@ abstract class LambdaAuthorizer extends Authorizer implements IAuthorizer {
     }
 
     this.restApiId = restApi.restApiId;
+
+    let functionName;
+
+    if (this.handler instanceof lambda.Function) {
+      // if not imported, extract the name from the CFN layer to reach
+      // the literal value if it is given (rather than a token)
+      functionName = (this.handler.node.defaultChild as lambda.CfnFunction).functionName;
+    } else {
+      // imported, just take the function name.
+      functionName = this.handler.functionName;
+    }
+
+    let authorizerToken;
+
+    if (!Token.isUnresolved(functionName)) {
+      authorizerToken = JSON.stringify({ functionName });
+    }
```

Comment:
You can pass tokens to `addToLogicalId()`, why do we check for them here?

Reply:
I have copied the existing logic from [here](https://github.com/aws/aws-cdk/blob/32781f825352f9cb43d8fed5c122b454275b3076/packages/%40aws-cdk/aws-apigateway/lib/integrations/lambda.ts#L91)

I am guessing it is because the token can change which could cause the logical ID to change, even though the function name hadn't changed

Reply:
The other logic did not use this method, and `addToLogicalId()` has a particular way of dealing with tokens. Please remove this token check and just pass it through to `addToLogicalId()`.

Reply:
I think the other logic I linked to is used in the `addToLogicalId()` call [here](https://github.com/aws/aws-cdk/blob/4f83e02b85229ebdff3f32ba6fd662ffd707d8db/packages/%40aws-cdk/aws-apigateway/lib/method.ts#L239), however I have reworked it to handle unresolved tokens